x-pack/plugins/fleet/server/services/agent_policy.ts

```diff
@@ -102,19 +103,49 @@ import { packagePolicyService } from './package_policy';
 import { incrementPackagePolicyCopyName } from './package_policies';
 import { outputService } from './output';
 import { agentPolicyUpdateEventHandler } from './agent_policy_update';
-import { escapeSearchQueryPhrase, normalizeKuery } from './saved_object';
+import { escapeSearchQueryPhrase, normalizeKuery as _normalizeKuery } from './saved_object';
 import { getFullAgentPolicy, validateOutputForPolicy } from './agent_policies';
 import { auditLoggingService } from './audit_logging';
 import { licenseService } from './license';
 import { createSoFindIterable } from './utils/create_so_find_iterable';
 import { isAgentlessEnabled } from './utils/agentless';
 import { validatePolicyNamespaceForSpace } from './spaces/policy_namespaces';
+import { getSettings } from './settings';
 
-const SAVED_OBJECT_TYPE = AGENT_POLICY_SAVED_OBJECT_TYPE;
+const SAVED_OBJECT_TYPE = LEGACY_AGENT_POLICY_SAVED_OBJECT_TYPE;
 
 const KEY_EDITABLE_FOR_MANAGED_POLICIES = ['namespace'];
 
+function normalizeKuery(savedObjectType: string, kuery: string) {
+  if (savedObjectType === LEGACY_AGENT_POLICY_SAVED_OBJECT_TYPE) {
+    return _normalizeKuery(
+      savedObjectType,
+      kuery.replace(
+        new RegExp(`${AGENT_POLICY_SAVED_OBJECT_TYPE}\\.`, 'g'),
+        `${savedObjectType}.attributes.`
+      )
+    );
+  } else {
+    return _normalizeKuery(
+      savedObjectType,
+      kuery.replace(
+        new RegExp(`${LEGACY_AGENT_POLICY_SAVED_OBJECT_TYPE}\\.`, 'g'),
+        `${savedObjectType}.attributes.`
+      )
+    );
+  }
+}
+
 class AgentPolicyService {
+  private async getSavedObjectType() {
```

Comment:
This is the part where we add the most complexity with an opt-in migration, storing this in a saved object means we have to query the SO each time we query a space aware saved object.